pkg/dynamiclistener/server.go

```diff
@@ -404,6 +424,21 @@ func (s *Server) getCertificate(hello *tls.ClientHelloInfo) (*tls.Certificate, e
 	return tlsCert, nil
 }
 
+func (s *Server) getRotateSettingInDays() (int64, error) {
+	setting, err := s.settingsLister.Get("", settings.RotateCertsIfExpiringInDays.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return 0, err
```

Comment:
I feel like we should default to 7 in case of an error. And log an error

Reply:
Added the logs